daffodil-core/src/main/scala/org/apache/daffodil/compiler/Compiler.scala

```diff
@@ -283,7 +284,31 @@ class Compiler private (var validateDFDLSchemas: Boolean,
 
   def reload(savedParser: java.nio.channels.ReadableByteChannel): DFDL.DataProcessor = {
     try {
-      val objInput = new ObjectInputStream(new GZIPInputStream(Channels.newInputStream(savedParser))) {
+      val is = Channels.newInputStream(savedParser)
+
+      // Read the magic number and version information for this saved parser
+      // directly from the input stream. That information is not compressed or
+      // serialized
+      val magicNumber = "DAFFODIL "
+      magicNumber.foreach { c =>
+        if (is.read() != c.toInt) throw new InvalidParserException("The saved parser is only compatible with an older version of Daffodil")
+      }
+
+      val sb = new StringBuilder()
+      var byte: Int = -1
+      while ({ byte = is.read(); byte > 0 }) {
+        sb.append(byte.toChar)
+      }
+      if (byte == -1) {
+        throw new InvalidParserException("The saved parser file is not the correct format")
```

Comment:
This is worse than "not the correct format" it means the file is corrupted, ends early because at the point where we should get a NUL we instead get EOF.  At this point it has a proper prefix string, but doesn't have a proper version number and NUL separating it from the rest. This could say exactly that. Saved parser file has proper prefix 'DAFFODIL" but otherwise is not properly formed.

Reply:
I guess I was avoiding giving too much information that doesn't really help anything but to confuse a user. No one should be manually modifying  these files, so telling someone specifically what is wrong doesn't really help them fix the problem. If anything, they'll wonder what it means to be missing a prefix and try to fix that problem. The problem is really that the file was corrupted or something (we don't really know). Maybe instead of "not correct format' we sould say "looks to be corrupted"? Though, I'm not sure if that gives any extra information aside from guessing what cause the issue.

Reply:
Note, we also used this same "not the correct format" string below if we get a `ZipException`. And we use "not a valid parser" if there is a `StreamCorruptedException`. However we decided to describe these errors we should be consistent.

Reply:
Yeah, I think something like "corrupted" is fine.

Reply:
Ok, I've used "corrupted" and cleaned up the other error messages to improve consistency